QueueInteropTransport.php

```diff
@@ -217,6 +223,13 @@ private function setMessageMetadata(Message $interopMessage, Envelope $envelope)
 
     private function encodeMessage(Envelope $envelope): Message
     {
+        /** @var InteropMessageStamp $interopStamp */
+        $interopStamp = $envelope->last(InteropMessageStamp::class);
+
+        if ($interopStamp) {
+            return $interopStamp->getMessage();
+        }
```

Comment:
This method is called on `ack()`, `reject()` and `send()`. In the first two cases, there should *always* be an `InteropMessageStamp`. But for `send()` we would never look for this stamp - we would *always* want to create a new interop message. So, I think we should separate this into two different methods: `encodeMessage()` (which does what it did before) and `findInteropStamp()`, which holds this new logic - here's an example from the Doctrine transport - https://github.com/symfony/symfony/blob/b92e4ed9d4e480ecfb5dfc0f1dd588b283c7e0a6/src/Symfony/Component/Messenger/Transport/Doctrine/DoctrineReceiver.php#L130-L140

Reply:
Yeah this makes it more clean. Keeping with the naming of `encodeMessage()`, I just called it `findMessage()`, since it returns the `Message` directly instead of the `InteropMessageStamp`. Is this valid?